Runtime/AvatarCreator/Scripts/UI/Elements/TemplateSelectionElement.cs

```diff
@@ -31,15 +36,52 @@ private void Awake()
         /// </summary>
         public async void LoadAndCreateButtons()
         {
-            await LoadTemplateData();
-            CreateButtons(avatarTemplates.ToArray(), async (button, asset) =>
+            await LoadAndCreateButtons(false);
+        }
+        
+        /// <summary>
+        /// Asynchronously loads avatar template data and creates button elements for each template.
+        /// Each button is created with an icon fetched from the template's image URL.
+        /// </summary>
+        public async Task LoadAndCreateButtons(bool useCachedResponse)
+        {
+            if (!useCachedResponse || avatarTemplates == null)
+            {
+                await LoadTemplateData();
+            }
+
+            var filteredTemplates = avatarTemplates!.Where(template => HasCorrectTemplateVersion(template) && HasCorrectGender(template)).ToList();
```

Comment:
I would still think we should remove the gender filter out of here. So then it will always create all the gendered template buttons, and then you also wouldn't need the useCachedResponse logic there. 
Instead add a line inside the CreateButtons function callback to set the visibility button visibility based on the gender. 
EG
```cs
                button.SetIcon(texture);
                button.gameObject.SetActive(asset.Gender == gender);
```

We should also then probably make a function that enables you to switch between male only, female only and all gender outfits. 
```cs
        public void SetVisibilityFilter(OutfitGender gender)
        {
            this.gender = gender;
            foreach (var avatarTemplate in avatarTemplates)
            {
                var button = GetButton(avatarTemplate.Id);
                if(button == null) continue;
                button.gameObject.SetActive(HasCorrectGender(avatarTemplate));
            }
        }
```